### PR TITLE
Parsing of `seed_transpiler` in `generate_preset_pass_manager`

### LINKED ISSUE #12938: Overflow error: Seed Transpiler Validation
### Environment

- **Qiskit version**: 1.1.1
- **Python version**: 3.11
- **Operating system**: MacOS

### What is happening?

The pass manager crashes when i am provided with negative `seed_transpiler` value to the for layout method is 'sabre' even for normal circuit. 
The seed_transpiler is not validated properly.
The issue is also  found when run with complex circuit with more number of two qubit gates.

### How can we reproduce the issue?

```python 
from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
from qiskit import QuantumCircuit
from qiskit_ibm_runtime.fake_provider import FakeSherbrooke
qc = QuantumCircuit(3)
qc.x(0)
qc.x(1)
qc.x(2)
qc.measure_all()

backend = FakeSherbrooke()
options = {
  'layout_method': 'sabre',
  'seed_transpiler': -1,
}

pass_manager = generate_preset_pass_manager(backend=backend, optimization_level=i, **options)
transpiled_circuit = pass_manager.run(qc)

print(backend.run(transpiled_circuit, shots=100).result().get_counts())
```

### What should happen?

There should be an unsigned int overflowerror
```bash
Traceback (most recent call last):
  File "/Users/xxx/Projects/circuit_generator/gen_mod.py", line 36, in <module>
    transpiled_circuit = pass_manager.run(qc)
                         ^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passmanager.py", line 440, in run
    return super().run(circuits, output_name, callback, num_processes=num_processes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passmanager.py", line 463, in wrapper
    return meth(*meth_args, **meth_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passmanager.py", line 225, in run
    return super().run(
           ^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/passmanager.py", line 231, in run
    out = [
          ^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/passmanager.py", line 232, in <listcomp>
    _run_workflow(program=program, pass_manager=self, callback=callback, **kwargs)
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/passmanager.py", line 292, in _run_workflow
    passmanager_ir, final_state = flow_controller.execute(
                                  ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/base_tasks.py", line 218, in execute
    passmanager_ir, state = next_task.execute(
                            ^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/base_tasks.py", line 218, in execute
    passmanager_ir, state = next_task.execute(
                            ^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/basepasses.py", line 195, in execute
    new_dag, state = super().execute(
                     ^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/passmanager/base_tasks.py", line 98, in execute
    ret = self.run(passmanager_ir)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passes/layout/sabre_layout.py", line 269, in run
    components = disjoint_utils.run_pass_over_connected_components(dag, target, inner_run)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passes/layout/disjoint_utils.py", line 51, in run_pass_over_connected_components
    return [run_func(dag, cmap_components[0])]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/xxx/Projects/circuit_generator/venv/lib/python3.11/site-packages/qiskit/transpiler/passes/layout/sabre_layout.py", line 397, in _inner_run
    (initial_layout, final_permutation, sabre_result) = sabre_layout_and_routing(
                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^
OverflowError: can't convert negative int to unsigned
```

### Any suggestions?

Validation of seed_transpiler value in `generate_preset_pass_manager` function

### PR DESCRIPTION
**Pull request by jschuhmac**
### Summary
Parsing of `seed_transpiler` argument of `generate_preset_pass_manager` raises a `ValueError` if negative value is provided for the seed.

### Details and comments
Fixes #12938.
Parsing function for `seed_transpiler` added to parsing other non-target dependent pass manager options.  
The added test checks that `ValueError` is indeed raised if seed takes negative value.

### CODE REVIEW

File: qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py [hunk 1]
```diff
@@ -532,3 +533,11 @@ def _parse_approximation_degree(approximation_degree):
     if approximation_degree < 0.0 or approximation_degree > 1.0:
         raise TranspilerError("Approximation degree must be in [0.0, 1.0]")
     return approximation_degree
+
+
+def _parse_seed_transpiler(seed_transpiler):
+    if seed_transpiler is None:
+        return None
+    if seed_transpiler < 0:
```
Comment (1ucian0) [inline qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py]: maybe a good occasion to also make sure it is an integer? Now, `'seed_transpiler': 0.1`  fails at Sabre calltime.

File: releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Raise a `ValueError` if :meth:`.generate_preset_pass_manager` is provided
```
Comment (1ucian0) [inline releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml]:
```suggestion
    Raise a `ValueError` exception if :meth:`.generate_preset_pass_manager` is provided
```

File: qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py (hunk 1 above)
Comment (jschuhmac) [reply qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py]: Implemented in the [commit](https://github.com/Qiskit/qiskit/pull/12980/commits/2456df8b127efdda82e08d30643b21e6b854af31) below.

File: releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml (hunk 2 above)
Comment (jschuhmac) [reply releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml]: Implemented in the [commit](https://github.com/Qiskit/qiskit/pull/12980/commits/2456df8b127efdda82e08d30643b21e6b854af31) below.

File: releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    Raise a `ValueError` exception if :meth:`.generate_preset_pass_manager` is
+    not provided with a non-negative integer `seed_transpiler` argument.
```
Comment (ElePT) [inline releasenotes/notes/fix-negative-seed-pm-2813a62a020da115.yaml]:
```suggestion
    Fixed the behavior of :meth:`.generate_preset_pass_manager` to raise a `ValueError` exception if not provided with a non-negative integer `seed_transpiler` argument.
```

**Review by ElePT — COMMENTED** [review]
I added a minor suggestion for the release note wording to indicate the fix a bit clearer, other than than LGTM. This PR reminds me that we could probably refactor the tests using `generate_preset_pass_manager` to be a bit easier to find/contribute to.

PS: you can directly click on "commit suggestion" just below the suggested snippets and this will automatically commit the suggested changes ("add suggestion to batch" allows you to group several suggestions in a single commit).

File: qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py [hunk 4]
```diff
@@ -532,3 +533,11 @@ def _parse_approximation_degree(approximation_degree):
     if approximation_degree < 0.0 or approximation_degree > 1.0:
         raise TranspilerError("Approximation degree must be in [0.0, 1.0]")
     return approximation_degree
+
+
+def _parse_seed_transpiler(seed_transpiler):
+    if seed_transpiler is None:
+        return None
+    if not type(seed_transpiler) == int or seed_transpiler < 0:
```
Comment (ElePT) [inline qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py]: Lint is complaining about the type-checking method, see https://dev.azure.com/qiskit-ci/qiskit/_build/results?buildId=59920&view=logs&j=a15d9aca-d75c-55b1-7f76-889cfe884cf0&t=154f6391-7dd7-5733-2230-2f10f0798b76&l=618

I think Lint would prefer sth like:

```suggestion
    if not isinstance(seed_transpiler, int) or seed_transpiler < 0:
```

**Review by ElePT — APPROVED** [review]
Great, LGTM!! Thanks a lot @jschuhmac :)

**Pull request merged**